From: Hugo Lefeuvre <hugo.lefeuvre@neclab.eu>
Subject: Add tinyalloc entry to the menuconfig and initialize it
diff --git a/lib/ukboot/Config.uk b/lib/ukboot/Config.uk
index a888bc1c..694762fa 100644
--- a/lib/ukboot/Config.uk
+++ b/lib/ukboot/Config.uk
@@ -32,6 +32,16 @@ if LIBUKBOOT
 		  Satisfy allocation as fast as possible. No support for free().
 		  Refer to help in ukallocregion for more information.
 
+		config LIBUKBOOT_INITTINYALLOC
+		bool "tinyalloc"
+		select LIBTINYALLOC
+		help
+		  Minimalist allocator implementation, meant for use in systems
+		  with unmanaged linear memory such as WebAssembly or embedded
+		  systems. tinyalloc is highly configurable and offers high
+		  performance and reasonable memory usage when used and
+		  configured appropriately.
+
 		config LIBUKBOOT_NOALLOC
 		bool "No memory allocator"
 
diff --git a/lib/ukboot/boot.c b/lib/ukboot/boot.c
index 4e749aa5..b8bf0a20 100644
--- a/lib/ukboot/boot.c
+++ b/lib/ukboot/boot.c
@@ -45,6 +45,8 @@
 #include <uk/allocbbuddy.h>
 #elif CONFIG_LIBUKBOOT_INITREGION
 #include <uk/allocregion.h>
+#elif CONFIG_LIBUKBOOT_INITTINYALLOC
+#include <uk/tinyalloc.h>
 #endif
 #if CONFIG_LIBUKSCHED
 #include <uk/sched.h>
@@ -233,6 +235,8 @@ void ukplat_entry(int argc, char *argv[])
 			a = uk_allocbbuddy_init(md.base, md.len);
 #elif CONFIG_LIBUKBOOT_INITREGION
 			a = uk_allocregion_init(md.base, md.len);
+#elif CONFIG_LIBUKBOOT_INITTINYALLOC
+			a = uk_tinyalloc_init(md.base, md.len);
 #endif
 		} else {
 			uk_alloc_addmem(a, md.base, md.len);
